docs/cookbook/adjustDayOfMonth.mjs

```diff
@@ -0,0 +1,21 @@
+const date = Temporal.Date.from('2020-04-14');
+
+// Third day of next month:
+
+const thirdOfNextMonth = date.plus({ months: 1 }).with({ day: 3 });
+
+assert.equal(thirdOfNextMonth.toString(), '2020-05-03');
+
+// Last day of this month:
+
+const lastOfThisMonth = date.with({ day: 31 });
```

Comment:
Does `date.with({ day: Infinity })` work?  (Months could in theory have more than 31 days)

Another option:

```javascript
const lastOfThisMonth = date.plus({ months: 1 }).minus({ days: 1 });
```

Reply:
> Does `date.with({ day: Infinity })` work?

Yes, that does work in the polyfill currently, although I haven't checked the spec to see if it's intended to work.

> ```js
> const lastOfThisMonth = date.plus({ months: 1 }).minus({ days: 1 });
> ```

For an input of May 15, that will give June 14. Maybe this?

```js
date.plus({ months: 1 }).with({ day: 1 }).minus({ days: 1 })
```

Reply:
Sounds good.  Use either form, which ever you like more, or put them both into the cookbook

Reply:
I forgot about `day: date.daysInMonth` which would be more readable than either of those two. I'll go with that.